packages/windows/download_appveyor_build.rb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+
+require "pathname"
+require "veyor"
+require "octokit"
+
+def env_value(name)
+  value = ENV[name]
+  raise "Specify #{name} environment variable" if value.nil?
+  value
+end
+
+client = Octokit::Client.new(:access_token => env_value("GITHUB_TOKEN"))
+
+groonga_repository = "groonga/groonga"
+
+base_dir = Pathname.new(__FILE__).expand_path.dirname
+top_dir = base_dir.parent.parent
+base_version = (top_dir + "base_version").read.strip
+tag_name = "v#{base_version}"
+
+appveyor_url = "https://ci.appveyor.com/"
+appveyor_info = nil
+client.statuses(groonga_repository, tag_name).each do |status|
+  next unless status.target_url.start_with?(appveyor_url)
+  case status.state
+  when "pending"
+  # Ignore
+  else
+    match_data = /\/([^\/]+?)\/([^\/]+?)\/builds\/(\d+)\z/.match(status.target_url)
+    appveyor_info = {
+      account: match_data[1],
+      project: match_data[2],
+      build_id: match_data[3],
+    }
+    break
+  end
+end
+if appveyor_info.nil?
+  raise "No AppVeyor build"
+end
+
+start_build = appveyor_info[:build_id].to_i + 1
+build_history = Veyor.project_history(account: appveyor_info[:account],
+                                      project: appveyor_info[:project],
+                                      start_build: start_build,
+                                      limit: 1)
+build_version = build_history["builds"][0]["buildNumber"]
+project = Veyor.project(account: appveyor_info[:account],
+                        project: appveyor_info[:project],
+                        version: build_version)
+project["build"]["jobs"].each do |job|
+  job_id = job["jobId"]
+  artifacts = Veyor.build_artifacts(job_id: job_id)
+  arch = ""
+  suffix = ""
+  case job["name"]
+  when /ARCH=amd64/
+    arch = "x64"
+  when /ARCH=x86/
+    arch = "x86"
+  end
+  suffix << arch
+  case job["name"]
+  when /VS_VERSION=15/
+    suffix << "-vs2017"
+  when /VS_VERSION=14/
+    suffix << "-vs2015"
+  when /VS_VERSION=12/
+    suffix << "-vs2013"
+  end
+  artifacts.each do |artifact|
+    file_name = artifact["fileName"]
+    url = "#{appveyor_url}api/buildjobs/#{job_id}/artifacts/#{file_name}"
+    file_name  = artifact["fileName"].sub(/#{arch}/, suffix)
```

Comment:
`sub(/#{Regexp.escape(arch)}/, ...)` or `sub(arch, ...)`

Reply:
`local_file_name = file_name.sub(...)`

Reply:
How about using `groonga-${VERSION}-${ARCHITECTURE}-${VISUAL_STUDIO_VERSION}.zip` for file name generated by CMake?

Reply:
It seems that good idea to contain VISUAL_STUDIO_VERSIOn in archive name. I'll change it.

Reply:
fixed to use "fileName" (download zip as is)